packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -235,6 +235,28 @@ export abstract class SharedObject<TEvent extends ISharedObjectEvents = ISharedO
         };
     }
 
+    /**
+     * {@inheritDoc (ISharedObject:interface).getGCData}
+     */
+    public getGCData(): IGCData {
+        this._isSummarizing = true;
+         // Create a SummarySerializer that will be used to serialize IFluidHandles in this object. SummarySerializer
+         // tracks the routes of handles that it serializes. These represent outbound routes to other Fluid object.
+        const serializer = new SummarySerializer(this.runtime.channelsRoutingContext);
+
+        // Take a snapshot of the object's data which will serialize all the handles.
+        // Note: Using the snapshotCore API during GC is a little expensive but its okay for now.
```

Comment:
I think it's better to reference a future issue that will track the work to make it more efficient and reference it here.
Also we need to have some readme.md or something explaining the overall flow - it might be hard for uneducated user looking at your PR (or code after it lands) to reason about overall flow, why it's done this way, etc.

Reply:
Created an issue and referenced it here.
Yeah, a README would be good. Both for overall GC and how getting outbound routes work in DDSs. Created an issue #4548 to track this.